pkg/api/v1/orchestrator/routes/statusValueKV.go

```diff
@@ -0,0 +1,93 @@
+package routes
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/metal-toolbox/conditionorc/internal/status"
+	"github.com/metal-toolbox/rivets/events/registry"
+	"github.com/nats-io/nats.go"
+	"github.com/pkg/errors"
+
+	rctypes "github.com/metal-toolbox/rivets/condition"
+)
+
+type statusValueKV interface {
+	publish(
+		facilityCode string,
+		conditionID uuid.UUID,
+		controllerID registry.ControllerID,
+		conditionKind rctypes.Kind,
+		newSV *rctypes.StatusValue,
+		create,
+		onlyTimestamp bool,
+	) error
+}
+
+// statusValue implements the statusValueKV interface
+type statusValue struct{}
+
+func initStatusValueKV() statusValueKV {
+	return &statusValue{}
+}
+
+func (s *statusValue) publish(
+	facilityCode string,
+	conditionID uuid.UUID,
+	controllerID registry.ControllerID,
+	conditionKind rctypes.Kind,
+	newSV *rctypes.StatusValue,
+	create,
+	onlyTimestamp bool,
+) error {
+	statusKV, err := status.GetConditionKV(conditionKind)
+	if err != nil {
+		return errors.Wrap(errPublishStatus, err.Error())
+	}
+
+	key := rctypes.StatusValueKVKey(facilityCode, conditionID.String())
+	// create
+	if create {
+		newSV.CreatedAt = time.Now()
+		if _, errCreate := statusKV.Create(key, newSV.MustBytes()); errCreate != nil {
+			return errors.Wrap(errPublishStatus, errCreate.Error())
+		}
+
+		return nil
+	}
+
+	// update
+	currEntry, err := statusKV.Get(key)
+	if err != nil && !errors.Is(err, nats.ErrKeyNotFound) {
+		return errors.Wrap(errPublishStatus, err.Error())
+	}
+
+	// only timestamp update takes the current value
+	var publishSV *rctypes.StatusValue
+	if onlyTimestamp {
```

Comment:
I agree that would be the ideal approach, the current API covers for the way the controller library is implemented.
We can hash this out in a separate discussion.

Reply:
I'll look into updating the Controller library to serialize its updates using the KV revision, so that they don't end up clobbering each other, as of now that is not very well implemented.  
Once that is in place, the API can be simplified to accept the whole object as is instead of separate timestamp update handling.